offload_vio/server_rx/video_decoder.cpp

```diff
@@ -0,0 +1,162 @@
+    auto nvvideoconvert_1 = gst_element_factory_make("nvvideoconvert", "nvvideoconvert1");
+
+    auto videoconvert_0 = gst_element_factory_make("videoconvert", "videoconvert0");
+    auto videoconvert_1 = gst_element_factory_make("videoconvert", "videoconvert1");
+
+    auto h265parse_0 = gst_element_factory_make("h265parse", "h265parse0");
+    auto h265parse_1 = gst_element_factory_make("h265parse", "h265parse1");
+
+    auto decoder_img0 = gst_element_factory_make("nvv4l2decoder", "decoder_img0");
+    auto decoder_img1 = gst_element_factory_make("nvv4l2decoder", "decoder_img1");
+
+    auto caps_filter_0 = gst_element_factory_make("capsfilter", "caps_filter_0");
+    auto caps_filter_1 = gst_element_factory_make("capsfilter", "caps_filter_1");
+
+    g_object_set(G_OBJECT(caps_filter_0), "caps", gst_caps_from_string("video/x-raw,format=GRAY8"), nullptr);
+    g_object_set(G_OBJECT(caps_filter_1), "caps", gst_caps_from_string("video/x-raw,format=GRAY8"), nullptr);
+
+    // create caps with width and height
+    auto caps_x265 = gst_caps_new_simple("video/x-h264", "stream-format", G_TYPE_STRING, "byte-stream", "alignment",
+                                         G_TYPE_STRING, "au", nullptr);
+    g_object_set(G_OBJECT(_appsrc_img0), "caps", caps_x265, nullptr);
+    g_object_set(G_OBJECT(_appsrc_img1), "caps", caps_x265, nullptr);
+    gst_caps_unref(caps_x265);
+
+    g_object_set(G_OBJECT(_appsrc_img0), "stream-type", 0, "format", GST_FORMAT_BYTES, "is-live", TRUE, nullptr);
+    g_object_set(G_OBJECT(_appsrc_img1), "stream-type", 0, "format", GST_FORMAT_BYTES, "is-live", TRUE, nullptr);
+
+    g_object_set(G_OBJECT(decoder_img0), "low-latency-mode", TRUE, nullptr);
+    g_object_set(G_OBJECT(decoder_img1), "low-latency-mode", TRUE, nullptr);
+
+    g_object_set(_appsink_img0, "emit-signals", TRUE, "sync", FALSE, nullptr);
+    g_object_set(_appsink_img1, "emit-signals", TRUE, "sync", FALSE, nullptr);
+
+    g_signal_connect(_appsink_img0, "new-sample", G_CALLBACK(cb_new_sample), this);
+    g_signal_connect(_appsink_img1, "new-sample", G_CALLBACK(cb_new_sample), this);
+
+    _pipeline_img0 = gst_pipeline_new("pipeline_img0");
+    _pipeline_img1 = gst_pipeline_new("pipeline_img1");
+
+    auto identity = gst_element_factory_make("identity", "identity");
+    g_object_set(G_OBJECT(identity), "dump", TRUE, nullptr);
+    g_object_set(G_OBJECT(identity), "signal-handoffs", TRUE, nullptr);
+
+    gst_bin_add_many(GST_BIN(_pipeline_img0), _appsrc_img0, h265parse_0, identity, decoder_img0, videoconvert_0,
+                     nvvideoconvert_0, caps_filter_0, _appsink_img0, nullptr);
+    gst_bin_add_many(GST_BIN(_pipeline_img1), _appsrc_img1, h265parse_1, decoder_img1, videoconvert_1, nvvideoconvert_1,
+                     caps_filter_1, _appsink_img1, nullptr);
+
+    // link elements
+    if (!gst_element_link_many(_appsrc_img0, decoder_img0, nvvideoconvert_0, videoconvert_0, caps_filter_0, _appsink_img0,
```

Comment:
Same for the decoder.

Reply:
@qinjunj I am assuming you also updated this part of the code?

Reply:
yes will do

Reply:
updated